docs/resource_sets/resource_sets.rst

```diff
@@ -0,0 +1,56 @@
+**********************************
+Resource sets and partial compile
```

Comment:
I would add a few more things to document:

* I would add an actual example. Maybe this example can build further on the small code snippet you already have.
    1. First, show the full model with a new Routers that belong to different resource set
    2. Then, show a partial model that updates a subset of the Routers
    3. It would also be nice if the example contains a shared resource (a resource that doesn't belong a resource set). Maybe we can use `std::AgentConfig` for that and explain that all Routers are managed by the same agent.
* Also explain that shared resource are never deleted by a partial compile and that a full compile is required to garbage-collect these shared resource.
* Also mention that we have support to perform scheduled full compiles.